src/internal/packager/sbom/viewer/compare.js

```diff
@@ -0,0 +1,173 @@
+        return
+    }
+
+    let leftJson = document.getElementById("leftJson").files[0];
+    let rightJson = document.getElementById("rightJson").files[0];
+
+    let leftReader = new FileReader();
+    leftReader.readAsText(leftJson);
+
+    leftReader.onload = function() {
+        try {
+            let leftData = JSON.parse(leftReader.result);
+            const leftMap = {}
+            leftData.artifacts.map(artifact => {
+                if (!leftMap[artifact.name]) { leftMap[artifact.name] = {} }
+                leftMap[artifact.name][artifact.version] = artifact
+            });
+
+            let rightReader = new FileReader();
+            rightReader.readAsText(rightJson);
+
+            rightReader.onload = function() {
+                try {
+                    let rightData = JSON.parse(rightReader.result);
+                    const rightMap = {}
+                    rightData.artifacts.map(artifact => {
+                        if (!rightMap[artifact.name]) { rightMap[artifact.name] = {} }
+                        rightMap[artifact.name][artifact.version] = artifact
+                    });
+
+                    let differences = [];
+                    rightData.artifacts.map(artifact => {
+                        if (!leftMap[artifact.name]) {
+                            artifact.zarfDiff = "Added"
+                            differences.push(artifact)
+                        } else if (!leftMap[artifact.name][artifact.version]) {
+                            artifact.zarfDiff = "Changed"
+                            oldVersion = Object.keys(leftMap[artifact.name])[0]
+                            artifact.version = oldVersion + " -> " + artifact.version
+                            differences.push(artifact)
+                        }
+                    });
+
+                    leftData.artifacts.map(artifact => {
+                        if (!rightMap[artifact.name]) {
+                            artifact.zarfDiff = "Removed"
+                            differences.push(artifact)
+                        }
+                    });
+
+                    loadDataTable(differences, artifactsTable)
+                } catch (e) {
+                    showModal("Unable to Compare", "You must select 2 Syft JSON files")
+                }
+            };
+        } catch (e) {
+            showModal("Unable to Compare", "You must select 2 Syft JSON files")
+        }
+    };
+}
+
+function loadDataTable(artifacts, dataTable) {
+    const transformedData = artifacts.map(artifact => {
+        return [
+            diff(artifact.zarfDiff),
+            artifact.type,
+            artifact.name,
+            artifact.version,
+            fileList(artifact.metadata),
+            (artifact.metadata && artifact.metadata.description) || "-",
+            ((artifact.metadata && artifact.metadata.maintainer) || "-").replace(/\u003c(.*)\u003e/, `&nbsp;|&nbsp;&nbsp;<a href="mailto:$1">$1</a>`),
```

Comment:
this feels kind of gross, might be nice to at least bring the first part of the string into another variable maybe?